server/controllers/user.controller.js

```diff
@@ -115,11 +138,146 @@ const UserController = function () {
+      const recipientEmail = req.body.email;
+
+      const senderEmail = "jgomez.cascarita@gmail.com"; //Ask Armando if we could use a fake email instead of ours (i.e. no-reply-cascarita@gmail.com)
+      const senderName = "Cascarita";
+      const type = "classic";
+
+      const authCode =
+        emailType === "passwordCode" ? await generateOTP() : null;
+
+      const emailTypes = {
+        passwordCode: {
+          subject: "Cascarita - Reset Your Password",
+          type: "classic",
+          emailContent: `
+        <html>
+          <body>
+            <p>Hello,</p>
+            <p>Use this code to reset your password:</p>
+            <h1><strong>${authCode}</strong></h1>
+            <p>Thank you!</p>
+          </body>
+        </html>`,
+        },
+        formLink: {
+          subject: "Cascarita - Please fill out this form",
+          emailContent: `
+        <html>
+          <body>
+            <p>Hello,</p>
+            <p>Please fill out the following form:</p>
+            <a href="${typeformLink}">Fill out the form</a>
+            <p>Thank you!</p>
+          </body>
+        </html>`,
+        },
+      };
+
+      const { subject, emailContent } = emailTypes[emailType] || {};
+
+      const body = JSON.stringify({
+        sender: { name: senderName, email: senderEmail },
+        to: [{ email: recipientEmail }],
+        type: type,
+        subject,
+        htmlContent: emailContent,
+      });
+
+      const response = await fetch(url, {
+        method: "POST",
+        headers: {
+          Accept: "application/json",
+          "Content-Type": "application/json",
+          "api-key": apiKey,
+        },
+        body,
+      });
+
+      const responseData = await response.json();
+
+      const newAuthCode = {
+        email: req.body.email,
+        code: await hashOtp(authCode),
+        attempts: 0,
+        start_date: new Date(),
+        expiration_date: new Date(new Date().getTime() + 15 * 60000),
+      };
+
+      await AuthCode.build(newAuthCode).validate();
+      await AuthCode.create(newAuthCode);
+
+      return res.status(200).json({ data: responseData });
+    } catch (error) {
+      console.error("failed to send email:", error);
+      next(error);
+    }
+  };
+
+  var verifyOTP = async function (req, res, next) {
+    try {
+      const otp = req.body.otp;
+
+      const response = await AuthCode.findAll({
+        limit: 1,
+        where: {
+          email: req.body.email,
+        },
+        order: [["start_date", "DESC"]],
+      });
+
+      const userOtp = response[0];
+
+      if (!userOtp) {
+        res.status(404);
+        throw new Error(`no entry was found with email: ${req.body.email}`);
+      }
+
+      if (userOtp.expiration_date < new Date()) {
+        res.status(401);
+        throw new Error("auth code has expired. generate a new code");
+      }
+
+      const result = await compareOtp(otp, userOtp.code);
+
+      if (!result) {
```

Comment:
I see we are storing attempts. Are there any best practices for how to limit the userOpt code entry attempts? Like should we stop it after `x` attempts? If so consider the following suggestion. 
```suggestion
      const MAX_OPT_ATTEMPTS = 5;
      if (!result || userOtp.attempts >= MAX_OPT_ATTEMPTS) {
```

Reply:
Added!